### PR TITLE
Updates InferenceService examples to use regular node affinities.

### PR DESCRIPTION
**Pull request by jmarhee**
Resolves https://github.com/coreweave/infra-pm/issues/91

### CODE REVIEW

**Comment by jmarhee** [comment]
Previous commit mostly to ID which examples updates and what that affinity declaration would look like-- these examples need other changes for the current API version in use (v1beta1 -- presumably the version that supports affinities, etc.). A note to myself is that the readme needs to be updated along with the accompanying documentation.

**Comment by jmarhee** [comment]
The only thing not made clear (and potentially outstanding for this PR) from the updated spec is how to handle parallelism (it's in the spec for InferenceService, but when applying, its is rejected if used-- both this and the use of affinities is not documented in kfserving, or not well enough to be found in some aggressive searching), but these examples now apply with the use of an affinity.

**Comment by salanki** [comment]
@cesarmesones please review

File: online-inference/custom-pytorch-aitextgen/aitextgen-inferenceservice.yaml [hunk 1]
```diff
@@ -1,26 +1,30 @@
...
+          memory: 10Gi
+    affinity:
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+          - matchExpressions:
+            - key: gpu.nvidia.com/class
+              operator: In
+              values:
+              - Tesla_V100_sxm
```
Comment (cesarmesones) [inline online-inference/custom-pytorch-aitextgen/aitextgen-inferenceservice.yaml:30]: must have been an old name, no _sxm

File: online-inference/custom-sentiment/sentiment-inferenceservice.yaml [hunk 2]
```diff
@@ -1,29 +1,33 @@
...
-              memory: 6Gi
+  predictor:
+    maxReplicas: 10
+    minReplicas: 0
+    containers:
+    - name: kfserving-container
+      image: coreweave/fastai-sentiment:4
+      env:
+      - name: STORAGE_URI
+        value: pvc://model-storage/sentiment
```
Comment (cesarmesones) [inline online-inference/custom-sentiment/sentiment-inferenceservice.yaml:16]: I get this error when trying this example, node affinity seems fine however

```
$ k logs -f sentiment-predictor-default-00001-deployment-6c97454b98-9tqgc -c storage-initializer
/usr/local/lib/python3.7/site-packages/ray/autoscaler/_private/cli_logger.py:61: FutureWarning: Not all Ray CLI dependencies were found. In Ray 1.4+, the Ray CLI, autoscaler, and dashboard will only be usable via `pip install 'ray[default]'`. Please update your install command.
  "update your install command.", FutureWarning)
[I 210811 15:12:28 initializer-entrypoint:13] Initializing, args: src_uri [/mnt/pvc/sentiment/] dest_path[ [/mnt/models]
[I 210811 15:12:28 storage:50] Copying contents of /mnt/pvc/sentiment/ to local
Traceback (most recent call last):
  File "/storage-initializer/scripts/initializer-entrypoint", line 14, in <module>
    kfserving.Storage.download(src_uri, dest_path)
  File "/usr/local/lib/python3.7/site-packages/kfserving/storage.py", line 81, in download
    (_GCS_PREFIX, _S3_PREFIX, _LOCAL_PREFIX, _HTTP_PREFIX))
Exception: Cannot recognize storage type for /mnt/pvc/sentiment/
'gs://', 's3://', 'file://', and 'http(s)://' are the current available storage type.
```

File: online-inference/image-classifier/service/classifier-inferenceservice.yaml [hunk 3]
```diff
@@ -1,42 +1,46 @@
...
-            cpu: 3
-            memory: 10Gi
-            nvidia.com/gpu: 1
+  predictor:
+    # Max one request processed at the same time per container (GPU)
+    minReplicas: 0 # Allow scale to zero
+    maxReplicas: 3
+    tensorflow:
+      # The PVC and path inside the PVC to the model. The path is what we put after /models/ in export_dir in the notebook.
+      storageUri: pvc://model-storage/inception/
```
Comment (cesarmesones) [inline online-inference/image-classifier/service/classifier-inferenceservice.yaml:14]: Also get a storage error here

```
$ k logs -f classifier-predictor-default-00001-deployment-6b66fc58d9-zzrx7 -c storage-initializer
/usr/local/lib/python3.7/site-packages/ray/autoscaler/_private/cli_logger.py:61: FutureWarning: Not all Ray CLI dependencies were found. In Ray 1.4+, the Ray CLI, autoscaler, and dashboard will only be usable via `pip install 'ray[default]'`. Please update your install command.
  "update your install command.", FutureWarning)
[I 210812 02:53:04 initializer-entrypoint:13] Initializing, args: src_uri [/mnt/pvc/inception/] dest_path[ [/mnt/models]
[I 210812 02:53:04 storage:50] Copying contents of /mnt/pvc/inception/ to local
Traceback (most recent call last):
  File "/storage-initializer/scripts/initializer-entrypoint", line 14, in <module>
    kfserving.Storage.download(src_uri, dest_path)
  File "/usr/local/lib/python3.7/site-packages/kfserving/storage.py", line 81, in download
    (_GCS_PREFIX, _S3_PREFIX, _LOCAL_PREFIX, _HTTP_PREFIX))
Exception: Cannot recognize storage type for /mnt/pvc/inception/
'gs://', 's3://', 'file://', and 'http(s)://' are the current available storage type.
```

**Review by cesarmesones — CHANGES_REQUESTED** [review]
Not sure if @salanki can shine some light, I've tried just about everything to try to debug this storage error, even trying with the original alpha examples, and same error. Maybe it was always there, perhaps something stupid I'm missing. Not sure if you are getting the same thing.

File: online-inference/custom-sentiment/sentiment-inferenceservice.yaml (hunk 2 above)
Comment (salanki) [reply online-inference/custom-sentiment/sentiment-inferenceservice.yaml:16]: Did you download the model https://docs.coreweave.com/compass/examples/custom-sentiment#download-the-model?

Comment (cesarmesones) [reply online-inference/custom-sentiment/sentiment-inferenceservice.yaml:16]: sure didn't 🤣

**Review by cesarmesones — APPROVED** [review]

File: online-inference/custom-basnet/basnet-inferenceservice.yaml [hunk 4]
```diff
@@ -1,28 +1,33 @@
...
+      ports:
+      - containerPort: 80
+        protocol: TCP
+      resources:
+        limits:
+          cpu: "3"
+          memory: 8Gi
+          nvidia.com/gpu: "1"
+        requests:
+          cpu: "0.5"
```
Comment (salanki) [inline online-inference/custom-basnet/basnet-inferenceservice.yaml:23]: Let's set this to 500m

**Pull request merged**